src/commands/dev/lib/dev-server.ts

```diff
@@ -887,27 +900,38 @@ async function findBuildMatch(
   requestPath: string
 ): Promise<BuildMatch | null> {
   for (const match of matches.values()) {
-    const {
-      builderWithPkg: { builder }
-    } = match;
-    if (typeof builder.shouldServe === 'function') {
-      const shouldServe = await builder.shouldServe({
-        entrypoint: match.src,
-        files,
-        requestPath
-      });
-      if (shouldServe) {
-        return match;
-      }
-    } else if (findAsset(match, requestPath)) {
-      // If there's no `shouldServe()` function, then look up if there's
-      // a matching build asset on the `match` that has already been built.
+    if (await shouldServe(match, files, requestPath)) {
       return match;
     }
   }
   return null;
 }
 
+async function shouldServe(
+  match: BuildMatch,
+  files: BuilderInputs,
+  requestPath: string
+): Promise<boolean> {
+  const {
+    builderWithPkg: { builder }
+  } = match;
+  if (typeof builder.shouldServe === 'function') {
+    const shouldServe = await builder.shouldServe({
+      entrypoint: match.src,
+      files,
+      requestPath
+    });
```

Comment:
Need to send `config` too

Reply:
Good call. Added in 636be0f0161febbbe690fadc8a5f8507cbed1c99.